src/LoanManager.ts

```diff
@@ -50,12 +71,134 @@ export class LoanManager extends AbstractContract {
         return this.getProducts(false);
     }
 
-    // public async getProduct(productId: number): Promise<ILoanProduct> {}
+    public async newEthBackedLoan(product: LoanProduct, weiAmount: Wei, userAccount: string, minRate?: Tokens): Promise<Transaction> {
+        const loanCount: number = await this.getLoanCount();
+        const gasEstimate: number = loanCount===0 ? NEW_FIRST_LOAN_GAS : NEW_LOAN_GAS;
+
+        let web3Tx: TransactionObject<void>;
+        if(!isLoanManagerPreMarginLoan(this.instance)) {
+            if(!minRate) {
+                throw new AugmintJsError('missing minRate in loanmanager!')
+            }
+            web3Tx = this.instance.methods.newEthBackedLoan(product.id, minRate.toString());
+        } else {
+            if(minRate) {
+                throw new AugmintJsError('unexpected minRate in loanmanager!')
+            }
+            web3Tx = this.instance.methods.newEthBackedLoan(product.id)
+        }
+
+        return new Transaction(this.ethereumConnection, web3Tx, {
+            gasLimit: gasEstimate,
+            from: userAccount,
+            value: weiAmount.amount
+        });
+    }
+
+    public async getLoansForAccount(userAccount: string): Promise<Loan[]> {
+        const loanManagerInstance:LoanManagerInstance = this.instance;
+        const chunkSize: number = isLoanManagerV0(loanManagerInstance) ? LEGACY_CONTRACTS_CHUNK_SIZE : CHUNK_SIZE;
+        const loanCount: number = await loanManagerInstance.methods
+            .getLoanCountForAddress(userAccount)
```

Comment:
it would be simpler and more efficient to fetch chunks like `Exchange#getOrder` does it.

also, use Promise.all to do requests in parallel, instead of sequentially await-ing.

Reply:
done
however doing parallel requests is not an option here, as we only know if there will be another page after we got the result of the previous one.